### ISSUE TITLE
Slow loading of HR recruitment page on the Portal

### ISSUE THREAD

**Issue by Harshvardhan80** [ColoredCow/portal#3113]
**Describe the task**
The HR recruitment page on the portal is experiencing significant performance issues, with page load times taking 10+ seconds. This slow loading makes it difficult and frustrating to use the page effectively.

**Steps to Reproduce:**

1. Access the HR recruitment page on the portal.
![Screenshot 2023-05-31 194505](https://github.com/ColoredCow/portal/assets/112154429/9d288e20-426f-4a3a-9e38-f5d640041bd2)

2. Observe the significant delay in page load times.

**Expected behaviour**
The HR recruitment page should load within a reasonable timeframe, preferably under a few seconds, to ensure smooth user interaction.

**References**
![Screenshot 2023-05-31 195223](https://github.com/ColoredCow/portal/assets/112154429/d8ef483f-502f-440c-98fd-d09a13acc2d5)

**Comment by Ayush8923** [comment]
**Solution to solve this problem:-**

As far as I can check there is a lot of queries are running when the HR recruitment page is open. In the hr_application table, if there is a 150 - 200 rows in that table so the minimum time of the page is 1 - 2 seconds. 
And If there is 10,000 + data in that table so the minimum time of the page is open is 14 sec. So the total queries 11101 queries are executed. I think there are also unnecessary queries we can remove the ones we don't need. So we need to optimize the `index` function of the ApplicationController.

**Comment by deepankar-sajwan** [comment]
@Ayush8923 as per the testing done is the portal it is tasking around `3-4 seconds` to load the `HR recruitment page`.  If the expected time for loading `HR recruitment page` then you can deploy the feature on the production

**Comment by Ayush8923** [comment]
> @Ayush8923 as per the testing done is the portal it is tasking around `3-4 seconds` to load the `HR recruitment page`. If the expected time for loading `HR recruitment page` will be between 3-4 seconds, then you can deploy the feature on the production

@deepankar-sajwan Yes. Now it is taking 2 -3 seconds which is correct as compared to earlier. We will do more code cleanup and optimization in it, so I think it will be better.

**Comment by deepankar-sajwan** [comment]
> Reference

@Ayush8923 you can proceed with the code deployment
